src/Parser/Mage/Frost/Modules/Traits/Whiteout.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+
+import {formatNumber} from 'common/format';
+import SPELLS from 'common/SPELLS';
+import {calculateAzeriteEffects} from 'common/stats';
+
+import TraitStatisticBox, { STATISTIC_ORDER } from 'Interface/Others/TraitStatisticBox';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import EnemyInstances from 'Parser/Core/Modules/EnemyInstances';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+
+// Enemy debuffs that provide the triple damage multiplier onto Ice Lance
+const SHATTER_EFFECTS = [
+  SPELLS.WINTERS_CHILL.id,
+  SPELLS.FROST_NOVA.id,
+  SPELLS.ICE_NOVA_TALENT.id,
+  SPELLS.GLACIAL_SPIKE_DAMAGE.id,
+  SPELLS.RING_OF_FROST_DAMAGE.id,
+];
+
+// The number of seconds of cooldown reduction given to Frozen Orb per Ice Lance
+const FO_REDUCTION_SEC = 0.5;
+
+/**
+ * See known issue below, this is a temporary fix
+ * To update, see https://ptr.wowhead.com/spell=137020/frost-mage
+ * AURA = 1 + (Damage modifier)
+ */
+const FROST_MAGE_AURA = 1 + (-0.24);
+
+/**
+ * Ice Lance deals an additional X1 (stacks and scales) damage and reduces the cooldown of Frozen Orb by 0.5 (does not stack or scale) sec.
+ *
+ * Known issue: X1 value in trait data is a flat value and does not apply class % aura, damage buffs, etc, and is not a separate damage event from Ice Lance.
+ * This makes it very difficult to determine what the actual damage gained from the trait is; see: https://github.com/WoWAnalyzer/WoWAnalyzer/issues/2088
+ */
+class Whiteout extends Analyzer {
+  static dependencies = {
+    enemies: EnemyInstances,
+    spellUsable: SpellUsable,
+    statTracker: StatTracker,
+  };
+
+  hadFingersProc = false;
+  bonusDamagePerIceLance = 0;
+  frozenOrbReductions = 0;
+  totalWhiteoutDamage = 0;
+
+  constructor(...args) {
+    super(...args);
+    if (!this.selectedCombatant.hasTrait(SPELLS.WHITEOUT.id)) {
+      this.active = false;
+      return;
+    }
+
+    const ranks = this.selectedCombatant.traitsBySpellId[SPELLS.WHITEOUT.id];
+    for (const rank of ranks) {
+      const [damage] = calculateAzeriteEffects(SPELLS.WHITEOUT.id, rank);
+      this.bonusDamagePerIceLance += damage;
+    }
+
+    this.bonusDamagePerIceLance *= FROST_MAGE_AURA;
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.ICE_LANCE.id) {
+      return;
+    }
+    if (this.spellUsable.isOnCooldown(SPELLS.FROZEN_ORB.id)) {
+      this.frozenOrbReductions += 1;
```

Comment:
We usually try to provide more infos for CDRs, not just how much it effectively reduced the CD but also how much was wasted. Just to show players what the trait/talent could've done.

Not sure if its usefull in this scenario or if its usually good to delay frozen orb. Tell me if you think its pointless information for the trait and I'll merge it, the rest looks good.

Reply:
The trait technically does not influence the APL of Frost mage at all; as in it's not worth using Frozen Orb earlier than you usually would (you normally use on CD anyway, but certain fights you hold it)  to get a few extra reductions in. 

However I am planning on an addition to this module anyway to estimate the Frozen Orb damage added via the trait. The math would look like this:

extraOrbs =  frozenOrbReducedCD / ORB_CD; // If I reduced 30 seconds total and FO is a 60s cd, I got 0.5 exta orbs)
// averageOrbDamage is the total damage provided by frozen orb over the fight divded by the number of casts
estimatedFrozenOrbDamage = averageOrbDamage * extraOrbs;

If that makes sense to add, I can commit that before the merge. Or I can put it in as a separate PR. If it doesn't make sense to add, then I would say just merge this one if approved :)

Reply:
The idea was mainly geared towards just trying to remove some of the difference in value of what is shown on the analyzer vs what sims estimate. For example, at 355 ilvl, sims show 400 estimated DPS but only taking the ice lance damage into account shows 200dps. Obviously some of that discrepancy can be accounted for with just fight mechanics, etc, but part of it is missing the frozen orb damage. I will be looking more into this to see if I can make it more accurate (with a new PR), so thanks for merging this one!